crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -192,18 +198,176 @@ impl event::HcP2pHandler for WrapEvtSender {
+                    ErrorRes { msg_id, .. } => {
+                        if let Some(resp) = pending.lock().unwrap().respond(msg_id) {
+                            let _ = resp.send(msg);
+                        }
+                    }
+                    CallRemoteReq {
+                        msg_id,
+                        to_agent,
+                        zome_call_params_serialized,
+                        signature,
+                    } => {
+                        let dna_hash = DnaHash::from_k2_space(&space);
+                        let resp = match evt_sender
+                            .call_remote(dna_hash, to_agent, zome_call_params_serialized, signature)
+                            .await
+                        {
+                            Ok(response) => CallRemoteRes { msg_id, response },
+                            Err(err) => ErrorRes {
+                                msg_id,
+                                error: format!("{err:?}"),
+                            },
+                        };
+                        let resp = crate::wire::WireMessage::encode_batch(&[&resp])?;
+                        if let Err(err) = kitsune
+                            .space(space)
+                            .await?
+                            .send_notify(from_peer, resp)
+                            .await
+                        {
+                            tracing::debug!(?err, "Error sending call remote response");
+                        }
+                    }
+                    CallRemoteRes { msg_id, .. } => {
+                        if let Some(resp) = pending.lock().unwrap().respond(msg_id) {
+                            let _ = resp.send(msg);
+                        }
+                    }
+                    RemoteSignalEvt {
+                        to_agent,
+                        zome_call_params_serialized,
+                        signature,
+                    } => {
+                        let dna_hash = DnaHash::from_k2_space(&space);
+                        // remote signals are fire-and-forget
+                        // so it's safe to ignore the response
+                        let _response = evt_sender
+                            .call_remote(dna_hash, to_agent, zome_call_params_serialized, signature)
+                            .await;
+                    }
+                }
+                HolochainP2pResult::Ok(())
+            });
+        }
+        Ok(())
+    }
+}
+
+impl kitsune2_api::KitsuneHandler for HolochainP2pActor {
+    fn create_space(
+        &self,
+        _space: kitsune2_api::SpaceId,
+    ) -> BoxFut<'_, kitsune2_api::K2Result<kitsune2_api::DynSpaceHandler>> {
+        Box::pin(async move {
+            let this: Weak<dyn kitsune2_api::SpaceHandler> = self.this.clone();
+            if let Some(this) = this.upgrade() {
+                Ok(this)
+            } else {
+                Err(kitsune2_api::K2Error::other("instance has been dropped"))
```

Comment:
```suggestion
                Err(kitsune2_api::K2Error::other("HolochainP2pActor instance has been dropped"))
```

Otherwise this could take me a while to trace down :)